operators/src/plot/histogram.rs

```diff
@@ -0,0 +1,841 @@
+use crate::engine::{
+    ExecutionContext, InitializedOperator, InitializedOperatorImpl, InitializedPlotOperator,
+    Operator, PlotOperator, PlotQueryProcessor, PlotResultDescriptor, QueryContext, QueryProcessor,
+    QueryRectangle, TypedPlotQueryProcessor, TypedRasterQueryProcessor, TypedVectorQueryProcessor,
+};
+use crate::error;
+use crate::error::Error;
+use crate::string_token;
+use crate::util::Result;
+use futures::future::BoxFuture;
+use futures::stream::BoxStream;
+use futures::{FutureExt, StreamExt, TryFutureExt};
+use geoengine_datatypes::collections::{FeatureCollection, FeatureCollectionInfos};
+use geoengine_datatypes::plots::{Plot, PlotData};
+use geoengine_datatypes::primitives::{
+    DataRef, FeatureDataRef, FeatureDataType, Geometry, Measurement,
+};
+use geoengine_datatypes::raster::{Pixel, RasterTile2D};
+use serde::{Deserialize, Serialize};
+use snafu::{ensure, OptionExt};
+use std::convert::TryFrom;
+
+pub const HISTOGRAM_OPERATOR_NAME: &str = "Histogram";
+
+/// A histogram plot about either a raster or a vector input.
+///
+/// For vector inputs, it calculates the histogram on one of its attributes.
+///
+pub type Histogram = Operator<HistogramParams>;
+
+/// The parameter spec for `Histogram`
+#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
+pub struct HistogramParams {
+    /// Name of the (numeric) attribute to compute the histogram on. Ignored for operation on rasters.
+    pub column_name: Option<String>,
+    /// The bounds (min/max) of the histogram.
+    pub bounds: HistogramBounds,
+    /// If the bounds are empty, it is derived from the square-root choice rule.
+    pub buckets: Option<usize>,
+    /// Whether to create an interactive output (`false` by default)
+    #[serde(default)]
+    pub interactive: bool,
+}
+
+string_token!(Data, "data");
+
+/// Let the bounds either be computed or given.
+#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
+#[serde(untagged)]
+pub enum HistogramBounds {
+    Data(Data),
+    Values { min: f64, max: f64 },
```

Comment:
these values are only interesting for drawing the plot, aren't they? Why does the operator need them?